attestation-service/src/token/ear_broker.rs

```diff
@@ -0,0 +1,496 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::*;
+
+use ear::{
+    Algorithm, Appraisal, Ear, ExtensionKind, ExtensionValue, Extensions, RawValue, VerifierID,
+};
+use kbs_types::Tee;
+use log::{debug, warn};
+use openssl::ec::{EcGroup, EcKey};
+use openssl::nid::Nid;
+use openssl::pkey::PKey;
+use serde::Deserialize;
+use serde_json::{json, Value};
+use serde_variant::to_variant_name;
+use shadow_rs::concatcp;
+use std::collections::{BTreeMap, HashMap};
+use std::path::Path;
+use std::sync::Arc;
+use time::{Duration, OffsetDateTime};
+use verifier::TeeEvidenceParsedClaim;
+
+use crate::policy_engine::{PolicyEngine, PolicyEngineType};
+use crate::token::DEFAULT_TOKEN_WORK_DIR;
+use crate::AttestationTokenBroker;
+
+use super::{COCO_AS_ISSUER_NAME, DEFAULT_TOKEN_DURATION};
+
+pub const DEFAULT_PROFILE: &str = "tag:github.com,2024:confidential-containers/Trustee";
+pub const DEFAULT_DEVELOPER_NAME: &str = "https://confidentialcontainers.org";
+
+const DEFAULT_POLICY_DIR: &str = concatcp!(DEFAULT_TOKEN_WORK_DIR, "/ear/policies");
+
+const RULES: [&str; 8] = [
+    "instance_identity",
+    "configuration",
+    "executables",
+    "file_system",
+    "hardware",
+    "runtime_opaque",
+    "storage_opaque",
+    "sourced_data",
+];
+
+#[derive(Deserialize, Debug, Clone, PartialEq)]
+pub struct TokenSignerConfig {
+    pub key_path: String,
+    pub cert_url: Option<String>,
```

Comment:
Please add
```
#[serde(default = "Option::default")]
```

This is for parse config when `key_path` is give while `cert_url` is not given. The same for following